src/snapred/backend/recipe/algorithm/LoadGroupingDefinition.py

```diff
@@ -0,0 +1,112 @@
+import pathlib
+
+from mantid.api import AlgorithmFactory, MatrixWorkspaceProperty, PropertyMode, PythonAlgorithm
+from mantid.kernel import Direction
+
+from snapred.backend.recipe.algorithm.MantidSnapper import MantidSnapper
+
+name = "LoadGroupingDefinition"
+
+
+class LoadGroupingDefinition(PythonAlgorithm):
+    """
+    This algorithm creates a grouping workspace from a grouping definition file.
+    inputs:
+        GroupingFilename: str -- path of an input grouping definition file (NEXUS, XML, or DiffCal/HDF5)
+        InstrumentName: str -- name of an associated instrument
+        InstrumentFilename: str -- path of an associated instrument definition file
+        InstrumentDonor: str -- Workspace to optionally take the associate instrument from
+        OutputWorkspace: str -- name of an output grouping workspace
+    """
+
+    def PyInit(self) -> None:
+        # declare properties
+        self.declareProperty(
+            "GroupingFilename",
+            defaultValue="",
+            direction=Direction.Input,
+            doc="Path of an input grouping definition file",
+        )
+        self.declareProperty(
+            "InstrumentName", defaultValue="", direction=Direction.Input, doc="Name of an associated instrument"
+        )
+        self.declareProperty(
+            "InstrumentFilename",
+            defaultValue="",
+            direction=Direction.Input,
+            doc="Path of an associated instrument definition file",
+        )
+        self.declareProperty(
+            MatrixWorkspaceProperty("InstrumentDonor", "", Direction.Input, PropertyMode.Optional),
```

Comment:
Does this need to be declared a `MatrixWorkspaceProperty`?  So far we've handled all workspace properties by their string names.

Reply:
I copied that aspect from [this PR](https://github.com/neutrons/SNAPRed/pull/73/files#diff-d5a5b47544d5bb130b0bc0b6f79bc6e7717c08a51ae6d202e2af3bc036afa4b0R17). Pete has explained that adds more safety to the code as it verifies that the string name indeed represents a workspace.  Also, I don't see how else we could specify that a property is optional.